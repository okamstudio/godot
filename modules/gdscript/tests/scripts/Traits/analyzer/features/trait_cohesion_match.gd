uses SomeTrait, OtherTrait
uses CTrait, BTrait, ATrait

trait SomeTrait:
	func func_one():
		print("function one")

trait OtherTrait:
	func func_two():
		print("function two")

trait ATrait:
	uses BTrait

trait BTrait:
	uses CTrait

trait CTrait:
	func func_three():
		print("function three")

func test():
	func_one()
	func_two()
	func_three()
	print("ok")
