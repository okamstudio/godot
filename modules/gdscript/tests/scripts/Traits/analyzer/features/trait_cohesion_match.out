GDTEST_OK
function one
function two
function three
ok
